Code/src/main/java/nl/utwente/group10/haskell/expr/Expr.java

```diff
@@ -26,6 +26,19 @@
     public abstract Type analyze(final Env env, final GenSet genSet) throws HaskellException;
 
     /**
+     * Analyzes the type tree and resolves the type for this usage of this expression, using an empty GenSet. This
+     * method can be used to call when analyzing the root of the expression tree (the first step in the type inference
+     * of an expression).
+     *
+     * @param env The current Haskell environment.
+     * @return The type for this usage of this expression.
+     * @throws HaskellException The type tree contains an application of an incompatible type.
+     */
```

Comment:
It could be an idea to make the other analyze() protected to avoid any confusion.